File: ColtSteele/recursiveRange.py
"""
Description: Given a number add up all the numbers up the number you were given
Input: single Int
Output: single Int all the numbers added up to the input number
"""

def recursiveRange(n):
    if n == 0:
        return 0

    return n + recursiveRange(n-1)


if __name__ == "__main__":
    print(recursiveRange(10)) # yeah its 55

File: Java/JavaJ/src/main/java/com/coltsteele/problems/recursiveRange.java
package com.coltsteele.problems;
/*
* accepts a number and adds up all the number from zero to the number you were given
* */
public class recursiveRange {
    public static void main(String[] args){
        System.out.println(recursiveRange.compute(10));
    }

    public static Integer compute(Integer number){
        if(number == 0){
            return 0;
        }

        return number + compute(number-1);
    }
}
